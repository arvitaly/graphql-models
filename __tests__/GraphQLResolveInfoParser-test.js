"use strict";
const graphql_1 = require("graphql");
const GraphQLResolveInfoParser_1 = require("./../GraphQLResolveInfoParser");
fdescribe("GraphQLResolveInfoParser", () => {
    it("parse all fields", () => {
        const parser = new GraphQLResolveInfoParser_1.default(getInfo(`query Q1{ 
        node{ 
            model1{ field1 }
            model2{
                ... on Model2{
                    field2
                } 
            }
        } 
    }`));
        expect(parser.getFields()).toMatchSnapshot();
    });
    it("get node fields", () => {
        const parser = new GraphQLResolveInfoParser_1.default(getInfo(`query Q1{ node{ model1{ field1 } } }`));
        expect(parser.getNodeFields()).toMatchSnapshot();
    });
    it("get query one fields", () => {
        const parser = new GraphQLResolveInfoParser_1.default(getInfo(`query Q1{ viewer{ model1{ field1, model2{ field2 } }} }`));
        expect(parser.getQueryOneFields()).toMatchSnapshot();
    });
    it("get query connection fields simple", () => {
        const parser = new GraphQLResolveInfoParser_1.default(getInfo(`query Q1{
            viewer{
                modelName1s{
                    edges{
                        node{
                            name
                        }
                    }
                    
                }
            }
        }`));
        expect(parser.getQueryConnectionFields()).toMatchSnapshot();
    });
    it("get query connection fields with fragments", () => {
        const parser = new GraphQLResolveInfoParser_1.default(getInfo(`query Q1{
            viewer{
                model1(first:10){
                    pageInfo{
                        hasNextPage
                    }
                    ...F1
                }}
            }
            fragment F1 on Model1Connection{
                 edges{
                     node{
                         ...F2
                     }
                 }
            }
            fragment F2 on Model1{
                model1{
                    field1
                    model2{
                        field2
                    }
                }
            }
            `));
        expect(parser.getQueryConnectionFields()).toMatchSnapshot();
    });
});
function getInfo(query) {
    const document = graphql_1.parse(query);
    let fragments = {};
    document.definitions.filter((definition) => {
        return definition.kind === "FragmentDefinition";
    }).map((node) => {
        fragments[node.name.value] = node;
    });
    const operation = document
        .definitions.find((node) => node.kind === "OperationDefinition");
    return {
        fieldNodes: operation.selectionSet.selections,
        operation,
        fragments,
    };
}
//# sourceMappingURL=GraphQLResolveInfoParser-test.js.map