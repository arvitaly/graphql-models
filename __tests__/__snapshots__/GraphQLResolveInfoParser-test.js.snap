exports[`GraphQLResolveInfoParser get node fields 1`] = `
Array [
  Object {
    "fields": Array [
      Object {
        "fields": Array [],
        "name": "field1",
      },
    ],
    "name": "model1",
  },
]
`;

exports[`GraphQLResolveInfoParser get query connection fields simple 1`] = `
Array [
  Object {
    "fields": Array [],
    "name": "name",
  },
]
`;

exports[`GraphQLResolveInfoParser get query connection fields with fragments 1`] = `
Array [
  Object {
    "fields": Array [
      Object {
        "fields": Array [],
        "name": "field1",
      },
      Object {
        "fields": Array [
          Object {
            "fields": Array [],
            "name": "field2",
          },
        ],
        "name": "model2",
      },
    ],
    "name": "model1",
  },
]
`;

exports[`GraphQLResolveInfoParser get query one fields 1`] = `
Array [
  Object {
    "fields": Array [],
    "name": "field1",
  },
  Object {
    "fields": Array [
      Object {
        "fields": Array [],
        "name": "field2",
      },
    ],
    "name": "model2",
  },
]
`;

exports[`GraphQLResolveInfoParser parse all fields 1`] = `
Array [
  Object {
    "fields": Array [
      Object {
        "fields": Array [
          Object {
            "fields": Array [],
            "name": "field1",
          },
        ],
        "name": "model1",
      },
      Object {
        "fields": Array [
          Object {
            "fields": Array [],
            "name": "field2",
          },
        ],
        "name": "model2",
      },
    ],
    "name": "node",
  },
]
`;
